src/app/course/course-question-snippet/course-question-snippet.component.ts

```diff
@@ -80,4 +93,24 @@ export class CourseQuestionSnippetComponent implements OnInit {
         }
         return '';
     }
+
+    switchFavorite(uqjId: number, favoriteStatus: boolean, questionId: number): void {
+        const data: { id: number, status: boolean } = {id: uqjId, status: !favoriteStatus};
```

Comment:
you could make this a data type if you are using it in multiple places

Reply:
Added a global variable for "data"